vignettes/stratigraphic-column.Rmd

```diff
@@ -0,0 +1,291 @@
+        axis.text.x = element_blank(),
+        axis.ticks.x = element_blank())
+```
+
+We can already see something that roughly resembles a stratigraphic column. However, there is a lot we can do to improve the aesthetics of our plot. For example, the column named `color` in our dataframe specifies the hexadecimal color corresponding to the dominant lithology of the unit. We can use this to color-code the units by lithology.
+
+```{r column_b, eval = TRUE, fig.height = 8}
+# Plot stratigraphic column
+ggplot(san_juan_units, aes(ymin = b_age, ymax = t_age, xmin = 0, xmax = 1)) +
+  # Plot units
+  geom_rect(alpha = 0.5, fill = san_juan_units$color, color = "black",
+            linetype = 2) +
+  # Reverse direction of y-axis
+  scale_y_reverse(limits = c(145, 66), n.breaks = 10, name = "Time (Ma)") +
+  # Theming
+  theme_classic() +
+  theme(legend.position = "none",
+        axis.line.x = element_blank(),
+        axis.title.x = element_blank(),
+        axis.text.x = element_blank(),
+        axis.ticks.x = element_blank())
+```
+
+Great! Now let's add labels indicating the names of the different units.
+
+```{r column_c, eval = TRUE, fig.height = 8}
+# Add midpoint age for plotting
+san_juan_units$m_age <- (san_juan_units$b_age + san_juan_units$t_age) / 2
+
+# Plot stratigraphic column
+ggplot(san_juan_units, aes(ymin = b_age, ymax = t_age, xmin = 0, xmax = 1)) +
+  # Plot units
+  geom_rect(alpha = 0.5, fill = san_juan_units$color, color = "black",
+            linetype = 2) +
+  # Add text labels
+  geom_text_repel(aes(x = 1, y = m_age, label = unit_name),
+                  size = 4, hjust = 0, force = 2,
+                  min.segment.length = 0, direction = "y",
+                  nudge_x = rep_len(x = c(2, 3), length.out = 17)) +
+  # Reverse direction of y-axis
+  scale_y_reverse(limits = c(145, 66), n.breaks = 10, name = "Time (Ma)") +
+  # Theming
+  theme_classic() +
+  theme(legend.position = "none",
+        axis.line.x = element_blank(),
+        axis.title.x = element_blank(),
+        axis.text.x = element_blank(),
+        axis.ticks.x = element_blank())
+```
+
+And finally, we can add an additional facet on the y-axis indicating the different stages of the Cretaceous, using the R package `deeptime`:
+
+```{r column_d, eval = TRUE, fig.height = 8}
+# Plot stratigraphic column
+ggplot(san_juan_units, aes(ymin = b_age, ymax = t_age, xmin = 0, xmax = 1)) +
+  # Plot units
+  geom_rect(alpha = 0.5, fill = san_juan_units$color, color = "black",
+            linetype = 2) +
+  # Add text labels
+  geom_text_repel(aes(x = 1, y = m_age, label = unit_name),
+                  size = 3.5, hjust = 0, force = 2,
+                  min.segment.length = 0, direction = "y",
+                  nudge_x = rep_len(x = c(2, 3), length.out = 17)) +
+  # Add additional unit rectangles
+  geom_rect(aes(ymin = b_age, ymax = t_age, xmin = 0.9, xmax = 1),
+            fill = san_juan_units$color, color = "black") +
```

Comment:
What's the point of these rectangles? It doesn't look bad, but it might be good to explain to the reader why we are adding these.

Reply:
Agreed!

Reply:
In the end I removed them as I don't think they were doing the job they were intended for.